projects/distributed-tracing/src/shared/dashboard/widgets/table/mode-toggle-table-widget.model.ts

```diff
@@ -0,0 +1,82 @@
+import { TableDataSource, TableMode, TableRow } from '@hypertrace/components';
+import { ArrayPropertyTypeInstance, ENUM_TYPE, ModelTemplatePropertyType } from '@hypertrace/dashboards';
+import { ARRAY_PROPERTY, Model, ModelJson, ModelProperty } from '@hypertrace/hyperdash';
+import { NEVER, Observable } from 'rxjs';
+import { SpecificationBackedTableColumnDef } from './table-widget-column.model';
+import { TableWidgetModel } from './table-widget.model';
+
+@Model({
+  type: 'mode-toggle-table-widget',
+  displayName: 'Mode Toggle Table Widget'
+})
+export class ModeToggleTableWidgetModel extends TableWidgetModel {
```

Comment:
Given that this wouldn't use most of the values from the parent, polymorphism may not make sense here. That is, it wouldn't make sense to define columns on a mode-toggle-table-widget which always gets columns from its delegate. If you do want to use polymorphism over unions, I think it may make sense to make an abstract `TableWidgetModel` defining the apis needed by the renderer, then have two sibling implementations: `ModeToggleTableWidgetModel` and `BasicTableWidgetModel`

Reply:
To that abstract class, we'd probably want to add a few more things (and to the delegate) - namely getters for things like searchAttribute and any other model properties that are accessed directly by the renderer.